hack/make-rules/test-cmd-util.sh

```diff
@@ -1608,7 +1608,7 @@ run_non_native_resource_tests() {
   kubectl "${kube_flags[@]}" delete foos test --cascade=true
 
   # Make sure it's gone
-  kube::test::get_object_assert foos "{{range.items}}{{$id_field}}:{{end}}" ''
+  kube::test::wait_object_assert foos "{{range.items}}{{$id_field}}:{{end}}" ''
```

Comment:
I'm not 100% sure whether the deletion policy is defaulting to foreground or background, but either way, and correct me if I'm wrong (I discussed this with @deads2k), with GC enabled there's no guarantee that the resource being deleted will be removed from etcd upon completion of the delete request; the GC has to remove a finalizer from the deletion target after processing dependents one way or another. So all the calls which delete and then assert the deletion target is gone should wait rather than assuming a synchronous deletion.